draft-packages/stories/Button.stories.tsx

```diff
@@ -331,3 +330,12 @@ export const MultipleButtons = () => (
     <Button label="Exit" automationId="demo-button-2" />
   </div>
 )
+
+export const CustomComponent = () => {
+  const customComponent = (props: CustomButtonProps) => (
+    <a href="#" {...props} />
```

Comment:
I was thinking that the risk here would be that if the props signature of the custom component doesn't match that of `CustomButtonProps`, some properties would get missed. But if all of our code is in in typescript, this error would be alerted, so I guess this would be fine.

We just need to hope that developers don't pass each prop one by one, otherwise if new props are added to `CustomButtonProps` may not get custom component instance.